test/e2e/upgrades/apps/deployments.go

```diff
@@ -98,89 +86,82 @@ func (t *DeploymentUpgradeTest) Setup(f *framework.Framework) {
+	rsClient := c.ExtensionsV1beta1().ReplicaSets(ns)
 
-	deployment, err := c.Extensions().Deployments(t.oldD.Namespace).Get(t.oldD.Name, metav1.GetOptions{})
+	deployment, err := deploymentClient.Get(deploymentName, metav1.GetOptions{})
 	framework.ExpectNoError(err)
 	t.updatedD = deployment
 
-	By(fmt.Sprintf("Waiting for deployment %q to complete adoption", deployment.Name))
-	framework.ExpectNoError(framework.WaitForDeploymentStatus(c, deployment))
+	By(fmt.Sprintf("Checking UID to verify deployment %q survives upgrade", deploymentName))
+	if t.updatedD.UID != t.oldD.UID {
+		framework.ExpectNoError(fmt.Errorf("expected same deployment UID: %v got: %v", t.oldD.UID, t.updatedD.UID))
+	}
 
-	By(fmt.Sprintf("Checking that replica sets for deployment %q are the same as prior to the upgrade", t.updatedD.Name))
-	_, allOldRSs, newRS, err := deploymentutil.GetAllReplicaSets(t.updatedD, c.ExtensionsV1beta1())
+	By(fmt.Sprintf("Verifying deployment %q does not create extra replicasets", deploymentName))
+	rsSelector := labels.Set{"test": "upgrade"}.AsSelector()
+	rsList, err := rsClient.List(metav1.ListOptions{LabelSelector: rsSelector.String()})
 	framework.ExpectNoError(err)
-	if newRS == nil {
-		By(t.spewDeploymentAndReplicaSets(newRS, allOldRSs))
-		framework.ExpectNoError(fmt.Errorf("expected a new replica set for deployment %q", t.updatedD.Name))
-	}
-	if newRS.UID != t.newRS.UID {
-		By(t.spewDeploymentAndReplicaSets(newRS, allOldRSs))
-		framework.ExpectNoError(fmt.Errorf("expected new replica set:\n%#v\ngot new replica set:\n%#v\n", t.newRS, newRS))
-	}
-	if len(allOldRSs) != 1 {
-		By(t.spewDeploymentAndReplicaSets(newRS, allOldRSs))
-		errString := fmt.Sprintf("expected one old replica set, got %d\n", len(allOldRSs))
-		for i := range allOldRSs {
-			rs := allOldRSs[i]
-			errString += fmt.Sprintf("%#v\n", rs)
-		}
-		framework.ExpectNoError(fmt.Errorf(errString))
-	}
-	if allOldRSs[0].UID != t.oldRS.UID {
-		By(t.spewDeploymentAndReplicaSets(newRS, allOldRSs))
-		framework.ExpectNoError(fmt.Errorf("expected old replica set:\n%#v\ngot old replica set:\n%#v\n", t.oldRS, allOldRSs[0]))
+	rss := rsList.Items
+	if len(rss) != 2 {
+		framework.ExpectNoError(fmt.Errorf("expected 2 replicaset, got %d", len(rss)))
 	}
+
+	By(fmt.Sprintf("Verifying revision of the deployment %q is still 2", deploymentName))
+	framework.ExpectNoError(framework.WaitForDeploymentRevision(c, deployment, "2"))
+
+	By(fmt.Sprintf("Waiting for deployment %q to complete adoption", deploymentName))
+	framework.ExpectNoError(framework.WaitForDeploymentCompletes(c, deployment))
+
+	// Verify the upgraded deployment is active by scaling up the deployment to deploymentScaleNum
+	By(fmt.Sprintf("Scaling up replicaset of deployment %q to %d", deploymentName, deploymentScaleNum))
+	_, err = framework.UpdateDeploymentWithRetries(c, ns, deploymentName, func(deployment *extensions.Deployment) {
+		*deployment.Spec.Replicas = deploymentScaleNum
```

Comment:
Need to make sure replicas != scale num

Reply:
No, we should make sure `replicas = deploymentScaleNum`. Refer replicaset upgrade test for more info.

Reply:
For the purpose of this test, deploymentScaleNum needs to be larger than replicas, otherwise this is not a scale up. I'd prefer scale num to be set with something like replicas+1 rather than const. Test could have been modified in the future (e.g. change # of replicas).